transport/tchannel/header_test.go

```diff
@@ -155,3 +155,75 @@ func TestReadHeadersFailure(t *testing.T) {
 	})
 	require.Error(t, err)
 }
+
+func TestWriteHeaders(t *testing.T) {
+	tests := []struct {
+		msg string
+		// the headers are serialized in an undefined order so the encoding
+		// must be one of the following
+		bytes          []byte
+		orBytes        []byte
+		headers        map[string]string
+		tracingBaggage map[string]string
+	}{
```

Comment:
add a test with 2 keys that are different case otherwise the sae, e.g., "A" and "a"